ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/util/P2PDumpManager.java

```diff
@@ -0,0 +1,135 @@
+      return;
+    }
+
+    final Path gossipMessagesPath = this.directory.resolve(GOSSIP_MESSAGES_DIR);
+    if (gossipMessagesPath.toFile().mkdirs()) {
+      LOG.info("{} directory has been created to save gossip messages.", GOSSIP_MESSAGES_DIR);
+    }
+    if (gossipMessagesPath.resolve(DECODING_ERROR_SUB_DIR).toFile().mkdirs()) {
+      LOG.info(
+          "{} directory has been created to save gossip messages with decoding errors.",
+          DECODING_ERROR_SUB_DIR);
+    }
+    if (gossipMessagesPath.resolve(REJECTED_SUB_DIR).toFile().mkdirs()) {
+      LOG.info(
+          "{}} directory has been created to save rejected gossip messages.", REJECTED_SUB_DIR);
+    }
+    if (this.directory.resolve(INVALID_BLOCK_DIR).toFile().mkdirs()) {
+      LOG.info(
+          String.format(
+              "%s directory has been created to save invalid blocks.", INVALID_BLOCK_DIR));
+    }
+  }
+
+  public void saveGossipMessageDecodingError(
+      final String topic, final String arrivalTimestamp, final Bytes originalMessage) {
+    if (!enabled) {
+      return;
+    }
+    final String fileName = String.format("%s_%s.ssz", arrivalTimestamp, topic);
+    final String identifiers = String.format("Topic: %s", topic);
+    final Path topicPath =
+        Path.of(GOSSIP_MESSAGES_DIR).resolve(DECODING_ERROR_SUB_DIR).resolve(topic);
+    checkTopicDirExists(topicPath);
+    saveBytesToFile(
+        "gossip message with decoding error",
+        identifiers,
+        topicPath.resolve(fileName),
+        originalMessage);
+  }
+
+  public void saveGossipRejectedMessageToFile(
+      final String topic, final String arrivalTimestamp, final Bytes decodedMessage) {
+    if (!enabled) {
+      return;
+    }
+    final String fileName = String.format("%s_%s.ssz", arrivalTimestamp, topic);
+    final String identifiers = String.format("Topic: %s", topic);
+    final Path topicPath = Path.of(GOSSIP_MESSAGES_DIR).resolve(REJECTED_SUB_DIR).resolve(topic);
+    checkTopicDirExists(topicPath);
+    saveBytesToFile(
+        "rejected gossip message", identifiers, topicPath.resolve(fileName), decodedMessage);
+  }
+
+  public void saveInvalidBlockToFile(
+      final UInt64 slot, final Bytes32 blockRoot, final Bytes blockSsz) {
+    if (!enabled) {
+      return;
+    }
+    final String fileName =
+        String.format("slot%s_root%s.ssz", slot, blockRoot.toUnprefixedHexString());
+    final String identifiers = String.format("Slot: %s, Block Root: %s", slot, blockRoot);
+    saveBytesToFile(
+        "invalid block", identifiers, Path.of(INVALID_BLOCK_DIR).resolve(fileName), blockSsz);
+  }
+
+  private void saveBytesToFile(
+      final String object,
+      final String identifiers,
+      final Path relativeFilePath,
+      final Bytes bytes) {
+    final Path path = directory.resolve(relativeFilePath);
+    try {
+      Files.write(path, bytes.toArray());
+      LOG.info("Saved {} bytes to file. {}. Location: {}", object, identifiers, relativeFilePath);
```

Comment:
this function probably not really needed, the formatting of the identifiers to get this far purely for logging is a little bit of an anti-pattern and Im not sure i see an advantage...

Reply:
either that or potentially the identifiers string is just redundant or something but its a bit of a curious thing to be string formatting for a potential log that may not be even needed.

Reply:
Okay I will remove the need for the identifiers because the path is quite descriptive anyway.